src/mlpack/methods/ann/dists/bernoulli_distribution_impl.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file bernoulli_distribution_impl.hpp
+ * @author Atharva Khandait
+ *
+ * Implementation of the Bernoulli distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_BERNOULLI_DISTRIBUTION_IMPL_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_BERNOULLI_DISTRIBUTION_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "bernoulli_distribution.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename DataType>
+BernoulliDistribution<DataType>::BernoulliDistribution() :
+    applyLogistic(true),
+    eps(1e-10)
+{
+  // Nothing to do here.
+}
+
+template<typename DataType>
+BernoulliDistribution<DataType>::BernoulliDistribution(
+    const DataType&& param,
+    const bool applyLogistic,
+    const double eps) :
+    logits(param),
+    applyLogistic(applyLogistic),
+    eps(eps)
+{
+  if (applyLogistic)
+    LogisticFunction::Fn(logits, probability);
+  else
+  {
+    probability = arma::mat(logits.memptr(), logits.n_rows,
+        logits.n_cols, false, false);
+  }
+}
+
+template<typename DataType>
+DataType BernoulliDistribution<DataType>::Sample() const
+{
+  DataType sample;
+  sample = arma::conv_to<DataType>::from(arma::randu<DataType>
+      (probability.n_rows, probability.n_cols) <= probability);
+
+  return sample;
+}
+
+template<typename DataType>
+double BernoulliDistribution<DataType>::LogProbability(
+    const DataType&& observation) const
+{
+  return arma::accu(arma::log(probability + eps) % observation +
```

Comment:
We could use ``trunc_log``.

Reply:
Yeah we can, I will change it, but I will still need eps for the denominator.

Reply:
We can just leave the the ``eps`` parameter as a part of the constructor.

Reply:
Okay then, I will use `trunc_log` and `eps` for denominator then.